code/modules/clothing/spacesuits/_spacesuits.dm

```diff
@@ -75,6 +75,10 @@
 	var/mob/living/carbon/human/user = src.loc
 	if(!user || !ishuman(user) || !(user.wear_suit == src))
 		return
+	if(!cell && thermal_on)
+		for(var/datum/action/action in actions)
+			if(istype(action, /datum/action/item_action/toggle_spacesuit))
+				action.Trigger()

```

Comment:
this is a very roundabout way to turn it off. Is there a reason you're doing this instead of just calling whatever proc the action calls on the spacesuit?

Reply:
I am doing it this way so the action icon is updated. if I called just the proc in here it would turn off, but not swap the icon, and I did not want to duplicate code linking to an icon or add a proc on the action button just to set the button to off.

I did try, three different ways as seen in commit history, but this was the best way in my opinion, trigger the same way the player would for consistency. And if the button is missing for some reason, snowflake armour, nothing will break.

Reply:
I think it would be better to move the handling to the suit, emit a signal when the state changes, and then have the action button listen for that signal and trigger an update of its icon imo.

Reply:
I redid it with signals now.

Reply:
Looks great :+1: